### DISCUSSION TITLE
How to migrate from the built-in postgres database to an external database without losing data.

### DISCUSSION THREAD

**Discussion by matifali** [coder/coder#7058] [General] [answered]
Is there a way to migrate to an external database from the built-in database without losing the data?

**Comment by kylecarbs** [comment, marked as answer]
The following steps should work:

1. Stop your Coder deployment.
2. Run `coder server postgres-builtin-serve` in a background terminal.
3. Run `coder server postgres-builtin-url` and copy its output command.
4. Run `pg_dump <connection-string>` to dump the database contents. Then restore that content to an external database, and start Coder pointing at the new one!

Let me know if something needs to be changed; happy to help!

**Reply by matifali** [reply]
Thanks, Kyle, for the detailed steps. It was swift to move to an external DB.

**Reply by matifali** [reply]
Moving these steps to our docs.

**Reply by matifali** [reply]
Available in docs at: https://coder.com/docs/v2/latest/admin/configure#migrating-from-the-built-in-database-to-an-external-database
